### PR TITLE
Feat/#628 core completed date

### PR DESCRIPTION
**Pull request by samrichca**
**Description of changes**

- add core completed date
- add core completed percentage

**Type of Change**

- [ ] Bug
- [ ] Refactor
- [x] New Feature
- [ ] Release Candidate

**Checklist before requesting review:**

- [x] Check branch (code change PRs go to `develop` not master)
- [x] Check copyrights for new files
- [x] Manual testing
- [x] Regression tests completed and passing (double check number of tests).
- [x] Spelling has been checked.
- [x] Updated swagger docs accordingly (check it's still valid)
- [x] N/A Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/submission/submission-to-clinical/stat-calculator.ts [hunk 1]
```diff
@@ -249,11 +271,11 @@ function noNeedToCalcCoreStat(
   return false;
 }
 
-function isValidCoreStatOverride(updatedCompletion: any): updatedCompletion is CoreCompletionStats {
+function isValidCoreStatOverride(updatedCompletion: any): updatedCompletion is CoreCompletionFields {
   return Object.entries(updatedCompletion).every(
     ([type, val]) =>
       Object.values(schemaNameToCoreCompletenessStat).find(field => type === field) &&
-      isNumber(val) &&
+      typeof val === 'number' &&
```
Comment (samrichca) [inline src/submission/submission-to-clinical/stat-calculator.ts:278]: got a deprecation warning about this

File: src/submission/submission-to-clinical/stat-calculator.ts [hunk 2]
```diff
@@ -43,9 +42,17 @@ type CoreClinicalSchemaName =
   | ClinicalEntitySchemaNames.FOLLOW_UP
   | ClinicalEntitySchemaNames.SPECIMEN;
 
+const getCoreCompletionPercentage = (fields: CoreCompletionFields) =>
+  mean(Object.values(fields || {})) || 0;
+
+const getCoreCompletionDate = (donor: Donor, percentage: number) =>
+  percentage === 1
+    ? donor.completionStats?.coreCompletionDate || donor.updatedAt || new Date().toDateString()
```
Comment (joneubank) [inline src/submission/submission-to-clinical/stat-calculator.ts:50]: Is the logic here that if the file is missing the core completion date, you will just use the most recent updatedAt date for the value?

Comment (samrichca) [reply src/submission/submission-to-clinical/stat-calculator.ts:50]: yup. should I skip that and use the current time instead?

**Review by joneubank — APPROVED** [review]

File: src/submission/submission-to-clinical/stat-calculator.ts (hunk 2 above)
Comment (samrichca) [reply src/submission/submission-to-clinical/stat-calculator.ts:50]: discussed on slack - no

**Pull request merged**